ddtrace/tracer.py

```diff
@@ -608,24 +611,16 @@ def write(self, spans):
         Send the trace to the writer to enqueue the spans list in the agent
         sending queue.
         """
-        if not spans:
+        if not spans or not self.enabled:
```

Comment:
Do we have a test that covers this change?

Reply:
Hmm it was meant to introduce any new functionality as it was already being done below. As per the other comment, I'll move it back to where it was originally.